### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by iLynette**
## Project requirements

- Install CanCanCan in your project.

- Add a role column to the users table. Remember to use a migration for this.

- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

- For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

- For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -28,6 +28,17 @@ def show
     @post = Post.find(params[:id])
   end
```
Comment (Mre55) [inline app/controllers/posts_controller.rb:31]: Kindly, use `CanCanCan` for Post authorization here in `Post Controller`. To do that you can include something like this at the top between lines 1 and 2: 
```
load_and_authorize_resource
```
You can read more [here](https://www.rubydoc.info/github/CanCanCommunity/cancancan/CanCan%2FControllerAdditions%2FClassMethods:load_and_authorize_resource).

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @team,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The README is well documented ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 
_Check the comments under the review._



## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (iLynette) [reply app/controllers/posts_controller.rb:31]: I have added this. Thank you @Mre55 ☺️

**Review by keroloussamy — APPROVED** [review]
Hi @iLynette, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**